stem/solver.py

```diff
@@ -1,10 +1,488 @@
+
+    """
+    desired_iterations: int = 10
+    max_radius_factor: float = 1.0
+    min_radius_factor: float = 0.1
+
+    @property
+    def strategy_type(self):
+        """
+        Returns the strategy type name of the arc length strategy
+
+        Returns:
+            - str: strategy type name
+
+        """
+        return "arc_length"
+
+
+@dataclass
+class LinearSolverSettingsABC(ABC):
+    """
+    Class containing information about the linear solver settings
+
+    Attributes:
+        - scaling (bool): if true, the system matrix will be scaled before solving the linear system of equations.\
+            Default value is False.
+
+    """
+    scaling: bool = False
+
+    @property
+    @abc.abstractmethod
+    def solver_type(self):
+        raise Exception("abstract class of linear solver settings is called")
+
+
+@dataclass
+class Amgcl(LinearSolverSettingsABC):
+    """
+    Class containing information about the amgcl linear solver settings
+
+    Inheritance:
+        - :class:`LinearSolverSettingsABC`
+
+    Attributes:
+        - tolerance (float): tolerance for the linear solver convergence criteria. Default value is 1e-6.
+        - max_iterations (int): maximum number of iterations for the linear solver. Default value is 1000.
+
+    """
+    tolerance: float = 1e-6
+    max_iterations: int = 1000
+
+    @property
+    def solver_type(self):
+        """
+        Property for returns the solver type name of the amgcl linear solver settings
+
+        Returns:
+            - str: solver type name
+
+        """
+        return "amgcl"
+
+
+@dataclass
+class TimeIntegration:
+    """
+    Class containing information about the time integration
+
+    Attributes:
+        - start_time (float): start time of the analysis
+        - end_time (float): end time of the analysis
+        - delta_time (float): initial time step
+        - reduction_factor (float): factor used to reduce the time step when the solution diverges
+        - increase_factor (float): factor used to increase the time step when the solution converges within the minimum\
+            number of iterations
+        - max_delta_time_factor (float): maximum time step factor, used to limit the time step increase. Default value\
+            is 1000.
+
+    """
+
+    start_time: float
+    end_time: float
+    delta_time: float
+
+    reduction_factor: float
+    increase_factor: float
+    max_delta_time_factor: float = 1000
+
+
+@dataclass
+class SolverSettings:
+    """
+    Class containing information about the time integration, builder, strategy, scheme and linear solver.
+
+    Attributes:
+        - solution_type (:class:`SolutionType`): solution type, quasi-static, K0-procedure or dynamic
+        - time_integration (:class:`TimeIntegration`): time integration settings
+        - rebuild_level (int): 2 if the lhs matrix is rebuilt at each non-linear iteration, 1 if the lhs matrix \
```

Comment:
you set defaults for convergence_criteria; reset disp; .... and not for rebuild level and prebuild_dynamics. why not? i would add also defaults for these with STEM in mind

Reply:
This is a dangerous practise i would say. Stem can contain non-linear stiffness matrixs, in this case, using a rebuild level <2 or prebuild_dynamics on true, will either crash or return non-reliable results>

But for linear analysis, using rebuild level <2 en prebuild dynamics speed up the calculation alot

Reply:
possibly we can rename it to: is_stiffness_matrix_constant and are_mass_and_damping_constant, to make it a bit more intuitive